multiuser/permission/che-multiuser-permission-workspace/src/main/java/org/eclipse/che/multiuser/permission/workspace/server/spi/jpa/MultiuserJpaWorkspaceDao.java

```diff
@@ -306,12 +311,72 @@ public void unsubscribe() {
 
     @Override
     public void onCascadeEvent(BeforeAccountRemovedEvent event) throws Exception {
+      boolean nonStoppedExists = false;
       for (WorkspaceImpl workspace :
           iterate(
               (maxItems, skipCount) ->
                   workspaceManager.getByNamespace(
                       event.getAccount().getName(), false, maxItems, skipCount))) {
-        workspaceManager.removeWorkspace(workspace.getId());
+        if (STOPPED.equals(workspace.getStatus())) {
+          workspaceManager.removeWorkspace(workspace.getId());
+          continue;
+        }
+
+        nonStoppedExists = true;
+
+        if (STOPPING.equals(workspace.getStatus())) {
+          // it's not possible to forcibly stop workspace. Continue check other and fail after this
+          // to retry
+          continue;
+        }
+
+        // workspace is RUNNING or STARTING. It's needed to stop them and remove after
+        String owner = workspace.getRuntime().getOwner();
+        if (owner.equals(EnvironmentContext.getCurrent().getSubject().getUserId())) {
```

Comment:
I wonder if it's a good idea to always stop workspace with SA? Maybe the current user does not have permissions to the corresponding namespace anymore and if so - it could help to avoid a situation which cannot be handled by Che Admin.

Reply:
@skabashnyuk WDYT?

Reply:
accepted

Reply:
It was just an assumption ) I'm not sure Che Server SA always has the corresponding permissions.
But it should, otherwise stopping by activity would not work I believe